ot/mpz-ot/src/actor/kos/sender.rs

```diff
@@ -0,0 +1,330 @@
+use std::collections::HashMap;
+
+use async_trait::async_trait;
+use futures::channel::{mpsc, oneshot};
+use futures_util::{stream::Fuse, SinkExt, StreamExt};
+use mpz_core::{Block, ProtocolMessage};
+use mpz_ot_core::kos::msgs::SenderPayload;
+use utils_aio::{
+    non_blocking_backend::{Backend, NonBlockingBackend},
+    sink::IoSink,
+    stream::IoStream,
+};
+
+use crate::{
+    actor::kos::{
+        into_kos_sink, into_kos_stream,
+        msgs::{ActorMessage, Message, TransferPayload, TransferRequest},
+    },
+    kos::{Sender, SenderError, SenderKeys},
+    CommittedOTReceiver, CommittedOTSenderShared, OTError, OTReceiver, OTSenderShared, OTSetup,
+};
+
+use super::SenderActorError;
+
+enum Command {
+    GetKeys(GetKeys),
+    SendPayload(PendingPayload),
+    Shutdown(Shutdown),
+}
+
+struct GetKeys {
+    id: String,
+    caller_response: oneshot::Sender<Result<SenderKeys, SenderError>>,
+}
+
+struct PendingPayload {
+    id: String,
+    payload: SenderPayload,
```

Comment:
/// Used to send back the `Result` to the caller of the `PendingPayload` command.